### PR TITLE
Support vscode install directory variable and resolve in env var collection variables

### LINKED ISSUE #115053: Support ${vscodeInstallPath} or similar in environment variable collection
As a means to workaround the install directory changing frequently in insiders, js-debug its bootloader into workspace storage. This falls apart when rebuilding containers though as js-debug expects the file to be there but it's not, and it cannot fix the problem because the extension has not activated yet. More context: https://github.com/microsoft/vscode-remote-release/issues/4308

The solutions is to allow task variables to be resolved in environment variable collection (I don't think they are currently), and then support a new one which allows the usage of VS Code's install path. This way the path to the actual bootloader would remain static under the extension's directory and does not need to be installed. The variable `${vscodeInstallPath}` would be resolved on terminal creation, for example:

```ts
collection.append('PATH', '${vscodeInstallPath}/extensions/git/bin');
collection.replace('USED_BY_GIT', '${vscodeInstallPath}/extensions/git/bin/auth.sh');
collection.replace('NODE_OPTIONS', 'require("${vscodeInstallPath}/ms-vscode.js-debug/bootloader.js")')
```

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #115053

### CODE REVIEW

**Comment by Tyriar** [comment]
@alexr00 I ended up landing on `appFolder` to align with `workspaceFolder`, and I don't think we actually need the extension path for now, plus there are some complications getting that in the ext host/remote.

File: src/vs/workbench/api/common/extHostDebugService.ts [hunk 1]
```diff
@@ -945,6 +945,9 @@ export class ExtHostVariableResolverService extends AbstractVariableResolverServ
 			getConfigurationValue: (folderUri: URI | undefined, section: string): string | undefined => {
 				return configurationService.getConfiguration(undefined, folderUri).get<string>(section);
 			},
+			getAppRoot: (): string | undefined => {
+				return env ? env['VSCODE_CWD'] : undefined;
```
Comment (alexr00) [inline src/vs/workbench/api/common/extHostDebugService.ts:949]: I thought that `appFolder` was just going to be the dirname of `execPath`. I'm not sure that `appFolder` communicates that this is the cwd of the the VS Code executable. How about `execCwd`?

Comment (Tyriar) [reply src/vs/workbench/api/common/extHostDebugService.ts:949]: exec lives in <install path>/bin/, we need <install path> such that we can get to the extensions path from it

Comment (Tyriar) [reply src/vs/workbench/api/common/extHostDebugService.ts:949]: Also "cwd" is ambiguous imo

Comment (Tyriar) [reply src/vs/workbench/api/common/extHostDebugService.ts:949]: And dirname appears to be used as the actual directory name, not the directory _path_.

Comment (Tyriar) [reply src/vs/workbench/api/common/extHostDebugService.ts:949]: Dirname, folder, root is used in the variables, it's a bit of a mess.

**Pull request merged**